github/github.go

```diff
@@ -0,0 +1,161 @@
+// Copyright OpenSSF Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/ossf/scorecard/v4/clients/githubrepo/roundtripper"
+	"github.com/ossf/scorecard/v4/log"
+)
+
+const (
+	baseRepoURL = "https://api.github.com/repos/"
+)
+
+// RepoInfo is a struct for repository information.
+type RepoInfo struct {
+	Repo      repo `json:"repository"`
```

Comment:
Curious - since `repo` struct is not exposed how does an external package access this nested struct?

Reply:
> Curious - since `repo` struct is not exposed how does an external package access this nested struct?

Great question! Really, you want to ideally discourage direct usage.
Constructing a client using `NewClient(ctx)` is going to provide a consumer with a properly populated `RepoInfo` struct.

Thinking about the context in which we're running this code, we've got two use cases:
- "production" -- this is running within the environment of a GitHub Action, where the event path and thereby the repository information will be present to unmarshal into `RepoInfo`
- testing -- this is where we need to ensure the developer environment/tests are scaffolded in a way that mimics the production env (`testdata/` / fixtures)

In both situations, I _think_ it's not strictly required to have these fields exported, but I'm open to it, if needed.

Reply:
Thanks for the explanation. Leaving it as-is LGTM.